share/models/change.py

```diff
@@ -189,6 +191,12 @@ def _create(self, save=True):
 
     def _update(self, save=True):
         self.target.change = self
+
+        new_type = self.change.pop('@type', None)
+        if new_type:
+            # TODO is this enough?
```

Comment:
I think there is a method on TypedObjects called `transition_type` or `recast_as` that will handle the conversion for you.